src/west/app/project.py

```diff
@@ -1757,13 +1761,67 @@ def handle_auto_cache(self, project):
             # Then clone the repository into the local cache.
             cache_dir_parent = Path(cache_dir).parent
             cache_dir_parent.mkdir(parents=True, exist_ok=True)
+            self.dbg(f'{project.name}: create auto-cache for {project.url} in {cache_dir}')
             project.git(
                 ['clone', '--mirror', '--', project.url, os.fspath(cache_dir)], cwd=cache_dir_parent
             )
             self.create_auto_cache_info(project, cache_dir)
         else:
-            # The local cache already exists. Sync it with remote.
-            project.git(['remote', 'update', '--prune'], cwd=cache_dir)
+            # helper function to check if git commands in the auto-cache run successful
+            def check_git(cmd) -> bool:
+                '''
+                Helper function to run git command in the auto-cache directory
+                with suppressed output. If the command was successful (returncode 0),
+                return 'True', otherwise False.
+                '''
+                p = project.git(
+                    cmd, cwd=cache_dir, capture_stdout=True, capture_stderr=True, check=False
+                )
+                return not p.returncode
+
+            # Check if the auto-cache is corrupt
+            corrupt = not check_git(['cat-file', '-e', 'HEAD'])
+
+            # Check if the revision is already contained in the auto-cache.
+            contains_rev = check_git(['cat-file', '-e', f'{project.revision}^{{commit}}'])
```

Comment:
This is a nice feature to speedup the update, it should be the default to return early if the revision (_not branch!_) is already present.